### ISSUE TITLE
. as additional mail extension

### ISSUE THREAD

**Issue by RogerSik** [mailcow/mailcow-dockerized#5130]
### Summary

Currently it is possible to use + as mail extension.

Example jon+vacation@example.org

It would be nice, if we could extend it with dot.  Because there sadly still sites which dont allow to use +. So it is possible to use the dot as workaround for these websites. 

Example jon.vacation@example.org

GMail allows something similar: https://support.google.com/mail/answer/7436150?hl=en

### Motivation

Some sites dont allow + as extension. So its possible to use . as a "workaround". The + extensions helps to filter easily the emails.

**Comment by master2die** [comment]
I agree with you that such an option in the UI would certainly not be wrong. Maybe you can create satisfaction for yourself by using filters.

Example:

```
# Rewrite dots in local part to +
require ["variables", "regex"];

if header :contains "To" "." {
    # Extract addresses from the To field
    extract :matches :raw "$To" "([^<,> ]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})" "addresses";

    # Search for addresses and replace dots in local part
    foreach :var "address" :in "addresses" {
        set "localpart" "${1}";
        set "domainpart" "${2}";

        if string :matches "${localpart}" "*.*" {
            set "new_localpart" "${localpart//./+}";
            set "new_address" "${new_localpart}@${domainpart}";
            replace :header "To" "${address}" "${new_address}";
        }
    }
}

```

After the filter is enabled, all emails sent to addresses with a dot in the local part will be automatically rewritten by replacing the dot with a plus sign. For example, the address "[abc.def@example.org](mailto:abc.def@example.org)" will be rewritten to "[abc+def@example.org](mailto:abc+def@example.org)". Addresses without a dot in the local part will remain unchanged, for example, "[abcdef@abc.example.org](mailto:abcdef@abc.example.org)" will not be changed to "abcdef@abc+example.org".

Please remember to make sure the Dovecot server is functioning properly after making changes to the filter, and that all emails are properly delivered since my code snippet is not tested properly.

**Comment by RogerSik** [comment]
Many thanks and sorry for answering soo late. @master2die 

Where i would put this filter? In the mailcow backend under filter?

Because in the backend I get the error "line 5: unknown command extract"

![25_15-37-36](https://github.com/mailcow/mailcow-dockerized/assets/165814/f2e67e23-bf7f-4f2b-98aa-808e1df472f0)

**Comment by phende** [comment]
I would also love to have this, or something similar. I am currently migrating to mailcow from another mail system where we have been using - (dash) as 'plus' addressing for the last 15 years so there are many such addresses in the wild.

Seeing that both postfix and dovecot are using the same configuration for this (recipient_delimiter in data/conf/postfix/extra.cf and data/conf/dovecot/extra.cf), it would be great if we could configure that in mailcow.conf and get it propagated to both. Note that recipient_delimiter accepts multiple characters (curently set to '+-' in my case). In th OP case it could be just '.'.

But maybe it is SOGo that does not have support for it? Is the 'In Subfolder'/'In Subject' option processed by SOGo, or dovecot?

By the way, awesome work team. Mailcow is outstanding.
